app/src/main/java/com/dsd/carcompanion/api/repository/VehicleRepository.kt

```diff
@@ -0,0 +1,34 @@
+package com.dsd.carcompanion.api.repository
+
+import android.util.Log
+import com.dsd.carcompanion.api.service.VehicleService
+import com.dsd.carcompanion.api.utils.ResultOf
+import retrofit2.Response
+
+class VehicleRepository(
+    private val vehicleService: VehicleService
+) {
+
+    private suspend fun <T> fetchDataFromApi(call: suspend () -> Response<T>, transform: suspend (T) -> Unit): ResultOf<Unit> {
```

Comment:
This name is too abstract, but since its only testing for now just change the name

Reply:
done